src/Command/ReformatCommand.php

```diff
@@ -0,0 +1,113 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command;
+
+use MongoDB\Driver\Command;
+use Nette\Utils\Json;
+use PHPStan\Command\ErrorFormatter\ErrorFormatter;
+use PHPStan\Command\ErrorFormatter\JsonErrorFormatter;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ReformatCommand extends \Symfony\Component\Console\Command\Command
+{
+	private const NAME = 'reformat';
+
+	/** @var string[] */
+	private $composerAutoloaderProjectPaths;
+
+	/**
+	 * @param string[] $composerAutoloaderProjectPaths
+	 */
+	public function __construct(array $composerAutoloaderProjectPaths)
+	{
+		parent::__construct();
+		$this->composerAutoloaderProjectPaths = $composerAutoloaderProjectPaths;
+	}
+
+	protected function configure(): void
+	{
+		$this->setName(self::NAME)
+			 ->setDescription('Read a previously generated analysis result in JSON format and converts it to a different format')
+			 ->setDefinition([
+				new InputOption('error-format', null, InputOption::VALUE_REQUIRED, 'Format in which to print the result of the analysis', 'table'),
+				new InputOption('memory-limit', null, InputOption::VALUE_REQUIRED, 'Memory limit for the run'),
+				new InputOption('xdebug', null, InputOption::VALUE_NONE, 'Allow running with XDebug for debugging purposes'),
+			 ]);
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int
+	{
+		$inceptionResult = CommandHelper::begin(
+			$input,
+			$output,
+			['.'],
+			null,
+			$input->getOption('memory-limit'),
+			null,
+			[],
+			null,
+			null,
+			$input->getOption('xdebug')
+		);
+
+		$errorOutput = $inceptionResult->getErrorOutput();
+		$container = $inceptionResult->getContainer();
+
+		$errorFormat = $input->getOption('error-format');
+
+		$errorFormatterServiceName = sprintf('errorFormatter.%s', $errorFormat);
+		if (!$container->hasService($errorFormatterServiceName)) {
+			$errorOutput->writeLineFormatted(sprintf(
+				 'Error formatter "%s" not found. Available error formatters are: %s',
+				 $errorFormat,
+				 implode(', ', array_map(static function (string $name): string {
+					 return substr($name, strlen('errorFormatter.'));
+				 }, $container->findServiceNamesByType(ErrorFormatter::class)))
+			 ));
+			return 1;
+		}
+
+		/** @var ErrorFormatter $errorFormatter */
+		$errorFormatter = $container->getService($errorFormatterServiceName);
+
+		$jsonString = stream_get_contents(STDIN);
+		$analysisResult = self::deserialize($jsonString);
+		return $errorFormatter->formatErrors($analysisResult, $inceptionResult->getStdOutput());
+	}
+
+	/**
+	 * inverse of @throws \Nette\Utils\JsonException
+	 *
+	 * @see JsonErrorFormatter
+	 */
+	protected static function deserialize(string $jsonString): AnalysisResult
```

Comment:
private

Reply:
Also, this should be moved to a separate class and tested in an integration way against JsonErrorFormatter.